tests/vanilla/subscribe.test.tsx

```diff
@@ -0,0 +1,123 @@
+import { describe, expect, it, vi } from 'vitest'
+import { subscribeWithSelector } from 'zustand/middleware'
+import { createStore } from 'zustand/vanilla'
+
+describe('subscribe()', () => {
```

Comment:
This is nearly an exact copy of `tests/subscribe.test.tsx`. It seems worthwhile to test this with `zustand/vanilla`, but it's likely unnecessary.

Reply:
If you think it's worth, we can do that.
Let's do it in `main`.

Reply:
Yes, I think it's worthwhile to have these tests because the `api` object with `subscribe`, `getState`, etc. could be modified after the `createStore` call and before `return useBoundStore` in `src/react.ts`.

Reply:
I made a PR to update the tests in main: #2244